radish/matcher.py

```diff
@@ -91,8 +97,26 @@ def match_step(sentence, steps):
 
             match = compiled.search(sentence, evaluate_result=False)
             argument_match = ParseStepArguments(match)
+            if match:
+                longest_group = get_longest_group(match.match.regs)
+            else:
+                longest_group = 0
 
         if match:
-            return StepMatch(argument_match=argument_match, func=func)
+            if len(sentence) == longest_group:
+                return StepMatch(argument_match=argument_match, func=func)
 
     return None
+
+
+def get_longest_group(regs):
+    if len(regs) < 1:
+        return 0
+
+    longest_group = regs[0][1]
+
+    for i in range(1, len(regs)):
```

Comment:
Yeah, you can still use `for reg in regs[1:]:` though which is much more pythonic.